src/utils/common/thread_executor.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _NIXL_THREAD_EXECUTOR_H
+#define _NIXL_THREAD_EXECUTOR_H
+
+#include <functional>
+#include <mutex>
+#include <atomic>
+#include <unordered_map>
+#include <chrono>
+#include <memory>
+#include <string>
+#include <boost/asio.hpp>
+#include "nixl_time.h"
+
+namespace nixl {
+
+/**
+ * @brief Task execution mode
+ */
+enum class TaskMode {
+    ONESHOT, ///< Execute task once and remove from queue
+    PERIODIC ///< Execute task periodically until unregistered
+};
+
+/**
+ * @brief Task information structure
+ */
+struct TaskInfo {
+    std::function<void()> task;
+    TaskMode mode;
+    std::chrono::microseconds interval;
+    std::chrono::steady_clock::time_point next_execution;
+    bool active;
+    std::string name;
+    std::shared_ptr<boost::asio::steady_timer> timer;
```

Comment:
We currently don't use boost directly in NIXL C++ code. We have standalone Asio available as vendored dependency that you can use instead ([example](https://github.com/ai-dynamo/nixl/blob/main/src/plugins/obj/obj_executor.h) from one of the plugins).